contentctl/objects/detection_tags.py

```diff
@@ -1,132 +1,151 @@
     observable: List[Observable] = []
-    message: Optional[str] = Field(...)
-    product: list[SecurityContentProductName] = Field(...,min_length=1)
+    message: str = Field(...)
+    product: list[SecurityContentProductName] = Field(..., min_length=1)
     required_fields: list[str] = Field(min_length=1)
-    
+
     security_domain: SecurityDomain = Field(...)
 
     @computed_field
     @property
-    def risk_severity(self)->RiskSeverity:
+    def risk_severity(self) -> RiskSeverity:
         if self.risk_score >= 80:
             return RiskSeverity('high')
         elif (self.risk_score >= 50 and self.risk_score <= 79):
             return RiskSeverity('medium')
         else:
             return RiskSeverity('low')
 
-
-    
     cve: List[Annotated[str, "^CVE-[1|2][0-9]{3}-[0-9]+$"]] = []
     atomic_guid: List[AtomicTest] = []
     drilldown_search: Optional[str] = None
 
-
     # enrichment
-    mitre_attack_enrichments: List[MitreAttackEnrichment] = Field([],validate_default=True)
-    confidence_id: Optional[PositiveInt] = Field(None,ge=1,le=3)
-    impact_id: Optional[PositiveInt] = Field(None,ge=1,le=5)
+    mitre_attack_enrichments: List[MitreAttackEnrichment] = Field([], validate_default=True)
+    confidence_id: Optional[PositiveInt] = Field(None, ge=1, le=3)
+    impact_id: Optional[PositiveInt] = Field(None, ge=1, le=5)
     # context_ids: list = None
-    risk_level_id: Optional[NonNegativeInt] = Field(None,le=4)
+    risk_level_id: Optional[NonNegativeInt] = Field(None, le=4)
     risk_level: Optional[RiskLevel] = None
-    #observable_str: str = None
+    # observable_str: str = None
     evidence_str: Optional[str] = None
 
     @computed_field
     @property
-    def kill_chain_phases(self)->list[KillChainPhase]:
+    def kill_chain_phases(self) -> list[KillChainPhase]:
+        # TODO (cmcginley): wait for feedback from Eric
         if self.mitre_attack_enrichments is None:
             return []
-        phases:set[KillChainPhase] = set()
+        phases: set[KillChainPhase] = set()
         for enrichment in self.mitre_attack_enrichments:
             for tactic in enrichment.mitre_attack_tactics:
                 phase = KillChainPhase(ATTACK_TACTICS_KILLCHAIN_MAPPING[tactic])
                 phases.add(phase)
         return sorted(list(phases))
-    
-    #enum is intentionally Cis18 even though field is named cis20 for legacy reasons
+
+    # enum is intentionally Cis18 even though field is named cis20 for legacy reasons
     @computed_field
     @property
-    def cis20(self)->list[Cis18Value]:
+    def cis20(self) -> list[Cis18Value]:
         if self.security_domain == SecurityDomain.NETWORK:
             return [Cis18Value.CIS_13]
         else:
             return [Cis18Value.CIS_10]
 
-    
     research_site_url: Optional[HttpUrl] = None
     event_schema: str = "ocsf"
+    # TODO (cmcginley): mappings should be fleshed out into a proper class
     mappings: Optional[List] = None
```

Comment:
I believe this should be a list of of `dict[str, str]` yes? Perhaps we should make a model for `Mapping` since it has consistent key structure?

Reply:
ABSOLUTELY. Anything we can do to move away from `dicts` is a huge win in my book.
I will leave this one open as I haven't made that change yet.

Reply:
See https://github.com/splunk/contentctl/issues/221